cfme/tests/infrastructure/test_providers.py

```diff
@@ -250,24 +250,38 @@ class TestProvidersRESTAPI(object):
     @pytest.yield_fixture(scope="function")
     def custom_attributes(self, rest_api, setup_a_provider):
         provider = rest_api.collections.providers[0]
-        attrs = provider.custom_attributes.action.add(
-            {"name": "ca_name1", "value": "ca_value1"}, {"name": "ca_name2", "value": "ca_value2"})
+        body = []
+        for _ in range(2):
+            uid = fauxfactory.gen_alphanumeric(5)
+            body.append({
+                'name': 'ca_name_{}'.format(uid),
+                'value': 'ca_value_{}'.format(uid)
+            })
+        attrs = provider.custom_attributes.action.add(*body)
 
         yield attrs
 
         try:
+            # custom attributes can be deleted by tests, just log warning
             provider.custom_attributes.action.delete(*attrs)
         except APIException:
             pass
 
     @pytest.mark.uncollectif(lambda: version.current_version() < '5.7')
     @pytest.mark.tier(3)
     @test_requirements.rest
-    def test_add_custom_attributes(self, custom_attributes):
-        """Test adding custom attributes to provider using REST API."""
+    def test_add_custom_attributes(self, rest_api, custom_attributes):
+        """Test adding custom attributes to provider using REST API.
+
+        Metadata:
+            test_flag: rest
+        """
         assert len(custom_attributes) == 2
```

Comment:
I think you should drop this check, or not use the literal '2' but a class constant here and in the `custom_attributes` fixture